storage/grpc_client.go

```diff
@@ -2370,3 +2382,78 @@ func checkCanceled(err error) error {
 
 	return err
 }
+
+func (w *gRPCWriter) initializeRetryConfig() {
+	if w.attempts == 0 {
+		w.attempts = 1
+
+		if w.settings.retry == nil {
+			w.settings.retry = defaultRetry
+		}
+
+		if w.settings.retry.shouldRetry == nil {
+			w.settings.retry.shouldRetry = ShouldRetry
+		}
+
+		w.invocationID = uuid.New().String()
+
+		if w.settings.retry.backoff != nil {
+			w.backoff.Multiplier = w.settings.retry.backoff.Multiplier
+			w.backoff.Initial = w.settings.retry.backoff.Initial
+			w.backoff.Max = w.settings.retry.backoff.Max
+		}
+	}
+}
+
+// shouldRetry determines if a retry is necessary and if so waits the appropriate
+// amount of time. It returns true if the error is retryable or the error to be
+// surfaced to the user if not.
+func (w *gRPCWriter) shouldRetry(ctx context.Context, err error) (bool, error) {
```

Comment:
Can we name this something else? Just confusing to conflate this with the ShouldRetry that only takes an err and returns a bool, especially since the backoff pause happens inside this func.

Reply:
Yeah, that's fair. I'll think of something better.